kernel_gateway/services/api/swagger.json

```diff
@@ -0,0 +1,564 @@
+                },
+                "codemirror_mode": {
+                    "type": "string",
+                    "description": "Codemirror mode.  Can be a string *or* an valid Codemirror mode object.  This defaults to the string from the `language` property."
+                },
+                "env": {
+                    "type": "object",
+                    "description": "A dictionary of environment variables to set for the kernel. These will be added to the current environment variables.",
+                    "additionalProperties": {
+                        "type": "string"
+                    }
+                },
+                "help_links": {
+                    "type": "array",
+                    "description": "Help items to be displayed in the help menu in the notebook UI.",
+                    "items": {
+                        "type": "object",
+                        "required": [
+                            "text",
+                            "url"
+                        ],
+                        "properties": {
+                            "text": {
+                                "type": "string",
+                                "description": "menu item link text"
+                            },
+                            "url": {
+                                "type": "string",
+                                "format": "URL",
+                                "description": "menu item link url"
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "Kernel": {
+            "description": "Kernel information",
+            "required": [
+                "id",
+                "name"
+            ],
+            "properties": {
+                "id": {
+                    "type": "string",
+                    "format": "uuid",
+                    "description": "uuid of kernel"
+                },
+                "name": {
+                    "type": "string",
+                    "description": "kernel spec name"
+                }
+            }
+        },
+        "Session": {
+            "description": "A session",
+            "type": "object",
+            "properties": {
+                "id": {
+                    "type": "string",
+                    "format": "uuid"
+                },
+                "notebook": {
+                    "type": "object",
+                    "properties": {
+                        "path": {
+                            "type": "string",
+                            "description": "path to notebook"
+                        }
+                    }
+                },
+                "kernel": {
+                    "$ref": "#/definitions/Kernel"
+                }
+            }
+        }
+    }
+}
```

Comment:
Following up on this: we should update the documentation for contributors about when to update the Swagger spec and how to regenerate the JSON from the YML.